athina/evals/ragas/ragas_evaluator.py

```diff
@@ -0,0 +1,80 @@
+
+from typing import Optional
+from athina.interfaces.athina import AthinaExperiment
+from athina.interfaces.model import Model
+import time
+from typing import Optional
+from athina.interfaces.result import EvalResult, EvalResultMetric
+from athina.interfaces.model import Model
+from athina.helpers.logger import logger
+
+from ..base_evaluator import BaseEvaluator
+from datasets import Dataset
+from ragas.llms import LangchainLLM
+from langchain.chat_models import ChatOpenAI
+from ragas import evaluate
+from athina.keys import OpenAiApiKey
+
+
+class RagasEvaluator(BaseEvaluator):
+    _model: str
+    _openai_api_key: str
+    _experiment: Optional[AthinaExperiment] = None
+
+    def __init__(
+        self,
+        openai_api_key: Optional[str] = None,
```

Comment:
Why do we need this as a constructor argument? All other LLM evaluators just get the key from `OpenAiApiKey.get_key()`.

Is there any case where `OpenAiApiKey.get_key()` is not sufficient?

Reply:
When this is called from worker, then we need to pass openai api key from worker